theano/gpuarray/c_code/spatialtf_sampler.c

```diff
@@ -0,0 +1,220 @@
+        cuda_exit( gpu_ctx->ctx );
+
+        PyErr_Format( PyExc_RuntimeError,
+                      "Could not create xdesc: %s",
+                      cudnnGetErrorString(err) );
+        return -1;
+    }
+
+    // In the input tensor, we must use its width and height, instead
+    // of the grid's width and height. The number of images and channels
+    // should be the same as the grid dimensions
+    const int input_num_images = (int) PyGpuArray_DIM( input, 0 );
+    const int input_num_channels = (int) PyGpuArray_DIM( input, 1 );
+    const int input_height = (int) PyGpuArray_DIM( input, 2 );
+    const int input_width = (int) PyGpuArray_DIM( input, 3 );
+
+    if ( input_num_images != num_images ||
+         input_num_channels != num_channels )
+    {
+        PyErr_Format( PyExc_RuntimeError,
+                      "Input should have %d images and %d channels, got %d images and %d channels.",
+                       num_images, num_channels, input_num_images, input_num_channels );
+        return -1;
+    }
+
+    err = cudnnSetTensor4dDescriptor( spatialtf_ctx.xdesc, tf, dt, num_images,
+        num_channels, input_height, input_width );
+
+    if ( err != CUDNN_STATUS_SUCCESS )
+    {
+        spatialtf_context_destroy( &spatialtf_ctx );
+        cuda_exit( gpu_ctx->ctx );
+
+        PyErr_Format( PyExc_RuntimeError,
+                      "Could not initialize xdesc: %s",
+                      cudnnGetErrorString(err) );
+        return -1;
+    }
+
+    err = cudnnCreateTensorDescriptor( &(spatialtf_ctx.ydesc) );
+
+    if ( err != CUDNN_STATUS_SUCCESS )
+    {
+        spatialtf_context_destroy( &spatialtf_ctx );
+        cuda_exit( gpu_ctx->ctx );
+
+        PyErr_Format( PyExc_RuntimeError,
+                      "Could not create xdesc: %s",
+                      cudnnGetErrorString(err) );
+        return -1;
+    }
+
+    err = cudnnSetTensor4dDescriptor( spatialtf_ctx.ydesc, tf, dt, num_images,
+        num_channels, height, width );
+
+    if ( err != CUDNN_STATUS_SUCCESS )
+    {
+        spatialtf_context_destroy( &spatialtf_ctx );
+        cuda_exit( gpu_ctx->ctx );
+
+        PyErr_Format( PyExc_RuntimeError,
+                      "Could not initialize ydesc: %s",
+                      cudnnGetErrorString(err) );
+        return -1;
+    }
+
+    if ( NULL == *output )
+    {
+        const size_t out_dims[4] = { num_images, num_channels, height, width };
+
+        *output = pygpu_zeros( 4, &(out_dims[0]), input->ga.typecode, GA_C_ORDER,
+            gpu_ctx, Py_None );
+
+        if ( NULL == *output )
+        {
+            spatialtf_context_destroy( &spatialtf_ctx );
+            cuda_exit( gpu_ctx->ctx );
+
+            PyErr_SetString( PyExc_MemoryError,
+                             "Could allocate memory for spatial transformer's grid sampler" );
+            return -1;
+        }
+    }
+    else
+    {
+        GpuArray_memset( &( (*output)->ga ), 0 );
```

Comment:
Here also, check that it works with just pygpu_empty and without memset. If so, we could skip calls to memset. If not we will keep the code as it is now.